src/MapViewController.swift

```diff
@@ -75,6 +122,46 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
       return tgViewController.tilt
     }
   }
+  
+  open var panEnabled = true
+  open var pinchEnabled = true
+  open var rotateEnabled = true
+  open var shoveEnabled = true
+  
+  weak open var singleTapGestureDelegate: MapSingleTapGestureDelegate?
+  weak open var doubleTapGestureDelegate: MapDoubleTapGestureDelegate?
+  weak open var longPressGestureDelegate: MapLongPressGestureDelegate?
+  weak open var panDelegate: MapPanGestureDelegate?
+  weak open var pinchDelegate: MapPinchGestureDelegate?
+  weak open var rotateDelegate: MapRotateGestureDelegate?
+  weak open var shoveDelegate: MapShoveGestureDelegate?
+
+  weak open var featureSelectDelegate: MapFeatureSelectDelegate? {
+    didSet {
+      tgViewController.mapViewDelegate = mapViewDelegatesExist() ? self : nil
```

Comment:
That actually is how I intended it to work- we want to register the internal TG `mapViewDelegate` if any of these external delegates or  `onSceneLoaded` exist because the presence of at least one means we need to receive TG's `mapViewDelegate` callbacks.

You make a good point about unnecessary linkages :) With that in mind, I've updated the code to register for `mapViewDelegate` at the beginning of the lifecycle like we do for `gestureDelegate`